qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -724,6 +723,133 @@ def find_ami430_with_name(ami430_name: str) -> AMI430:
             vals=Bool()
         )
 
+        self.ramp_mode = Parameter(
+            name="ramp_mode",
+            instrument=self,
+            get_cmd=None,
+            set_cmd=None,
+            vals=Enum("default", "simultaneous"),
+            initial_value="default",
+        )
+
+    def _get_measured_field_vector(self) -> FieldVector:
+        return FieldVector(
+            x=self._instrument_x.field(),
+            y=self._instrument_y.field(),
+            z=self._instrument_z.field(),
+        )
+
+    def ramp_simultaneously(self, setpoint: FieldVector, time: float) -> None:
+        """
+        Ramp all axes simultaneously to the given setpoint and in the given time
+
+        The method calculates and sets the required ramp rates per magnet
+        axis, and then initiates a ramp simultaneously on all the axes. The
+        trajectory of the tip of the magnetic field vector is thus linear in
+        3D space, from the current field value to the setpoint.
+
+        If ``block_during_ramp`` parameter is ``True``, the method will block
+        until all axes finished ramping.
+
+        It is required for all axis instruments to have the same units for
+        ramp rate and field, otherwise an exception is raised. The given
+        setpoint and time are assumed to be in those common units.
+
+        Args:
+            setpoint: ``FieldVector`` setpoint
+            time: time in which the setpoint field has to be reached on all axes
+
+        """
+        (
+            common_field_units,
+            common_ramp_rate_units,
+        ) = self._raise_if_not_same_field_and_ramp_rate_units()
+
+        self.log.debug(
+            f"Simultaneous ramp: setpoint {setpoint.repr_cartesian()} "
+            f"{common_field_units} in {time} {common_ramp_rate_units}"
+        )
+
+        # calculate new ramp rates based on time and setpoint
+
+        start_field = self._get_measured_field_vector()
+        self.log.debug(
+            f"Simultaneous ramp: start {start_field.repr_cartesian()} "
+            f"{common_field_units}"
+        )
+        self.log.debug(
+            f"Simultaneous ramp: delta {(setpoint - start_field).repr_cartesian()} "
+            f"{common_field_units}"
+        )
+
+        new_ramp_rates = self.calculate_ramp_rates_for(
```

Comment:
add a sumultaneous_ramp_rate parameter for the ramp rate along the vector, so the there's another way of doiojng this ramp which is:
```python
mode('simultaneous')
sumultaneous_ramp_rate(0.050) # T/s
cartesian(x,y,z)  # do the ramp
```

Reply:
@JouriB i've implemented this using:
```python
i3d.ramp_mode("simultaneous")
i3d.vector_ramp_rate(0.05)  # requires all axes to have same units!!!
i3d.cartesian(x,y,z)  # or any other parameter call that initiates a ramp
```

What do you think? does it work as expected?

Reply:
@JouriB tested and confirmed that all works perfectly!